app/namespace.go

```diff
@@ -574,6 +574,33 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
 						},
 					},
 				},
+				{
+					Name:    "retention",
+					Usage:   "The length of time a closed workflow will be preserved before deletion",
+					Aliases: []string{"r"},
+					Subcommands: []*cli.Command{{
+						Name:    "set",
+						Aliases: []string{"s"},
+						Usage:   "Set the length of time a closed workflow will be preserved before deletion per namespace",
+						Flags: []cli.Flag{
+							NamespaceFlag,
+							RetentionFlag,
+							RequestIDFlag,
+						},
+						Action: func(ctx *cli.Context) error {
+							retention := ctx.Int(RetentionFlagName)
+							n, err := c.getNamespace(ctx.String(NamespaceFlagName))
+							if err != nil {
+								return err
+							}
+							if int32(retention) == n.Spec.RetentionDays {
```

Comment:
[Nit] if there is no uint flag, might want to check that a negative value is not supplied

Reply:
uint was available thanks!

Reply:
I turned this into a uint flag, but it looks like it turns it into "0" if negative, so checking for negative.